app/src/main/java/com/shevchenkovtwo/homework/ui/fragments/MovieDetailFragment.kt

```diff
@@ -28,32 +31,26 @@ class MovieDetailFragment : Fragment() {
     }
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
-        super.onViewCreated(view, savedInstanceState)
-        movieDetailViewModel = ViewModelProvider(this).get(MovieDetailViewModel::class.java)
-        movieDetailViewModel?.let { movieViewModel ->
-            movieViewModel.getSelectedMovie().observe(viewLifecycleOwner, { movie ->
-                fragmentMovieDetailBinding?.let { binding ->
-                    initView(movie, binding)
-                    navigateBack(binding)
-                    if (checkData(movie)){
-                        Snackbar.make(view, getString(R.string.error_message), Snackbar.LENGTH_SHORT).show()
-                    }
```

Comment:
@emartynov
This is not really connected to selected movie, so you don't need to do it every time when new movie is selected

Reply:
Reworked